apps/examples/websocket/websocket_main.c

```diff
@@ -747,9 +749,17 @@ int websocket_main(int argc, char *argv[])
 			goto error_with_input;
 		}
 
-		strncpy(input->server_ip, argv[2], 20);
-		strncpy(input->server_port, argv[3], 8);
-		strncpy(input->path, argv[4], 32);
+		int addr_len = strlen(argv[2]);
+		int port_len = strlen(argv[3]);
+		int path_len = strlen(argv[4]);
+
+		if (addr_len > WEBSOCKET_EXAMPLE_ADDR_LEN || port_len > WEBSOCKET_EXAMPLE_PORT_LEN
+			|| path_len > WEBSOCKET_EXAMPLE_PATH_LEN) {
+			goto error_with_input;
```

Comment:
is it so critical, that you should terminate execution?
Why cannot we terminate the string to max length?

Reply:
I think it cause a side effect